akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala

```diff
@@ -197,4 +200,24 @@ object GraphStages {
     }
     override def toString: String = s"SingleSource($elem)"
   }
+
+  private[stream] final class FutureSource[T](val future: Future[T]) extends GraphStage[SourceShape[T]] {
+    ReactiveStreamsCompliance.requireNonNullElement(future)
+    val shape = SourceShape(Outlet[T]("future.out"))
+    val out = shape.out
```

Comment:
I left it loke that to show that it is not needed. 😉

-- 
Cheers,
√
On 7 Jan 2016 01:02, "Konrad Malawski" <notifications@github.com> wrote:

> In akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala
> <https://github.com/akka/akka/pull/19367#discussion_r49027179>:
>
> > @@ -197,4 +200,24 @@ object GraphStages {
> >      }
> >      override def toString: String = s"SingleSource($elem)"
> >    }
> > +
> > +  private[stream] final class FutureSource[T](val future: Future[T]) extends GraphStage[SourceShape[T]] {
> > +    ReactiveStreamsCompliance.requireNonNullElement(future)
> > +    val shape = SourceShape(Outlet[T]("future.out"))
> > +    val out = shape.out
>
> YAS - YetAnotherStyle [image: :wink:]
> Doesn't really save lines either hm.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/akka/akka/pull/19367/files#r49027179>.
>